test/UnitTest/SourceControl/Git/NameToInternalNameTest.cls

```diff
@@ -17,6 +17,9 @@ Method TestRegularClassNames()
 	do $$$AssertEquals(##class(Utils).NameToInternalName("cls\SourceControl\Git\DoesNotExist.cls"),"")
 	// Regular class that doesn't exist and we don't ignore non-existent classes
 	do $$$AssertEquals(##class(Utils).NameToInternalName("cls\SourceControl\Git\DoesNotExist.cls", 1, 0),"SourceControl.Git.DoesNotExist.CLS")
+	do $$$AssertEquals(##class(Utils).NameToInternalName("test\UnitTest\Git\DoesNotExist.cls", 1, 0),"UnitTest.Git.DoesNotExist.CLS")
+	do $$$AssertEquals(##class(Utils).NameToInternalName("foo\UnitTest\Foo\Git\DoesNotExist.cls", 1, 0),"UnitTest.Foo.Git.DoesNotExist.CLS")
+	do $$$AssertEquals(##class(Utils).NameToInternalName("foo\UnitTest\Foo\Git\DoesNotExist.foo", 1, 0),"UnitTest.Foo.Git.DoesNotExist.FOO")
```

Comment:
I think we need to cover more of the "unmatching" cases in unit tests.

Reply:
By unmatching cases do you mean negative test cases where we expect to not find a matching mapping?

Reply:
Yes.